cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -153,14 +152,12 @@ func NewSupervisor(logger *zap.Logger, configFile string) (*Supervisor, error) {
 		return nil, err
 	}
 
-	if err := s.loadConfig(configFile); err != nil {
-		return nil, fmt.Errorf("error loading config: %w", err)
-	}
-
-	if err := s.config.Validate(); err != nil {
+	if err := cfg.Validate(); err != nil {
```

Comment:
would it make sense to also move the validation into the new `Load` function introduced in this PR? 
This would make it easier to further decouple the supervisor from other components that also use the config, such as the `Commander` - if we already know the config is valid before calling `NewSupervisor`, we can use the config to create a `Commander` and inject that into the `Supervisor` via the `NewSupervisor` function.

Reply:
I added validation to `Load`. I wasn't sure if I should remove it from `NewSupervisor` but I think as long as it's possible to create config by hand (not using `Load`) it's better to have this.